pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -0,0 +1,126 @@
+syntax = "proto3";
+
+package nsmconnect;
+
+import "github.com/ligato/networkservicemesh/netmesh/model/netmesh/netmesh.proto";
+
+message Label {
+    map<string ,string> selector = 1;
+}
+
+message Metadata {
+    string name = 1;
+    string namespace = 2;
+    Label labels = 3;
+}
+
+message KernelInterface {
+   Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message VhostUserInterface{
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message MemInterface {
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message SRIOV {
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+enum InterfaceType {
+    not_specified_type = 0;
+    kernel_interface = 1;
+    vhost_user = 2;
+    memif = 3;
+    sriov = 4;
+  }
+
+enum InterfacePreference {
```

Comment:
This approach feels suspicious to me, I'd prefer to see a list of enums where order is based on the index.

Reply:
Also, using the Oneof approach listed, these enums may be removed entirely.